src/main/java/life/qbic/omero/BasicOMEROClient.java

```diff
@@ -71,120 +78,109 @@ private void connect(String hostname, int port, String username, String password
   }
 
   /**
    * Returns any map annotation data (key value pairs of metadata) of a given image
    *
    * @param imageID the ID of the image
    * @return A list of MapAnnotationData objects
-   * @throws DSOutOfServiceException
-   * @throws DSAccessException
-   * @throws ExecutionException
    */
-  public List<MapAnnotationData> fetchMapAnnotationDataForImage(long imageID)
-      throws DSOutOfServiceException, DSAccessException, ExecutionException {
-    List<MapAnnotationData> annotations = new ArrayList<>();
-    for (AnnotationData d : loadAnnotationsForImage(imageID, MapAnnotationData.class)) {
-      annotations.add((MapAnnotationData) d);
-    }
-    return annotations;
+  public List<MapAnnotationData> fetchMapAnnotationDataForImage(long imageID) {
+    return loadAnnotationsForImage(imageID, MapAnnotationData.class);
   }
 
-  private Collection<? extends AnnotationData> loadAnnotationsForImage(long imageID,
-      Class<? extends AnnotationData> type)
-      throws DSOutOfServiceException, DSAccessException, ExecutionException {
+  /**
+   *
+   * @param imageID the omero identifier for the desired image
+   * @param <T> the desired subclass of {@link AnnotationData}
+   * @return a List containing annotation data for the given image matching the desired class
+   */
+  private <T extends AnnotationData> List<T> loadAnnotationsForImage(long imageID, Class<T> type) {
     long userID = this.gateway.getLoggedInUser().getId();
+    ImageData image;
+    List<AnnotationData> annotations;
 
-    BrowseFacility browse = gateway.getFacility(BrowseFacility.class);
-    ImageData image = browse.getImage(this.ctx, imageID);
+    try {
+      BrowseFacility browse = gateway.getFacility(BrowseFacility.class);
+      image = browse.getImage(this.ctx, imageID);
+    } catch (DSOutOfServiceException dsOutOfServiceException) {
+      throw new RuntimeException("Error while accessing omero service: broken connection, expired session or not logged in", dsOutOfServiceException);
+    } catch (ExecutionException executionException) {
+      throw new RuntimeException("Task aborted unexpectedly.", executionException);
+    } catch (DSAccessException dsAccessException) {
+      throw new RuntimeException("Could not pull data from the omero server.", dsAccessException);
+    }
 
-    List<Long> userIds = new ArrayList<Long>();
+    List<Long> userIds = new ArrayList<>();
     userIds.add(userID);
 
-    List<Class<? extends AnnotationData>> types = new ArrayList<Class<? extends AnnotationData>>();
+    List<Class<? extends AnnotationData>> types = new ArrayList<>();
     types.add(type);
 
-    MetadataFacility metadata = gateway.getFacility(MetadataFacility.class);
-    List<AnnotationData> annotations = metadata.getAnnotations(ctx, image, types, userIds);
-
+    try {
+      MetadataFacility metadata = gateway.getFacility(MetadataFacility.class);
+      annotations = metadata.getAnnotations(ctx, image, types, userIds);
```

Comment:
```suggestion
      annotations = metadata.getAnnotations(ctx, image, types, null);
```
fetches all annotations independent of users - which is what I think we want to do here

Reply:
Then we can also remove the `userId` from the method signature?

Reply:
currently it's taken from the gateway/logged in user, but in my opinion you can remove it